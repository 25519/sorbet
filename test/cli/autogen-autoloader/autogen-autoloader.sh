#!/bin/bash
set -eu

rm -rf output
mkdir -p output

main/sorbet --silence-dev-message --stop-after=namer -p autogen-autoloader:output test/cli/autogen-autoloader/{foo,bar}.rb

for file in $(find output -type f | sort); do
  printf "\n--- %s\n" "$file"
  cat "$file"
done
