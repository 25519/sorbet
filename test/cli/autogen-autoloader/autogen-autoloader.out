No errors! Great job.

--- output/Foo.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Foo')

module Foo
end

Opus::Require.autoload_map(Foo, {
  Bar: TODO,
})

--- output/Foo/Bar.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Foo::Bar')

module Foo::Bar
end

Opus::Require.autoload_map(Foo::Bar, {
  Jazz: TODO,
  Quuz: TODO,
})

--- output/Foo/Bar/Jazz.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Foo::Bar::Jazz')

class Foo::Bar::Jazz < Foo::Bar::Quuz
end

Opus::Require.autoload_map(Foo::Bar::Jazz, {
  JazBaz: TODO,
})

Opus::Require.for_autoload(Foo::Bar::Jazz, "test/cli/autogen-autoloader/foo.rb")

--- output/Foo/Bar/Jazz/JazBaz.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Foo::Bar::Jazz::JazBaz')

class Foo::Bar::Jazz::JazBaz
end

Opus::Require.for_autoload(Foo::Bar::Jazz::JazBaz, "test/cli/autogen-autoloader/foo.rb")

--- output/Foo/Bar/Quuz.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Foo::Bar::Quuz')

class Foo::Bar::Quuz
end

Opus::Require.for_autoload(Foo::Bar::Quuz, "test/cli/autogen-autoloader/foo.rb")

--- output/Yabba.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Yabba')

module Yabba
end

Opus::Require.autoload_map(Yabba, {
  Dabba: TODO,
})

--- output/Yabba/Dabba.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Yabba::Dabba')

module Yabba::Dabba
end

Opus::Require.autoload_map(Yabba::Dabba, {
  Jazz: TODO,
  Quuz: TODO,
})

--- output/Yabba/Dabba/Jazz.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Yabba::Dabba::Jazz')

module Yabba::Dabba::Jazz
end

Opus::Require.autoload_map(Yabba::Dabba::Jazz, {
  JazBaz: TODO,
})

--- output/Yabba/Dabba/Jazz/JazBaz.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Yabba::Dabba::Jazz::JazBaz')

class Yabba::Dabba::Jazz::JazBaz
end

Opus::Require.for_autoload(Yabba::Dabba::Jazz::JazBaz, "test/cli/autogen-autoloader/bar.rb")

--- output/Yabba/Dabba/Quuz.rb
# frozen_string_literal: true
# DO NOT TOUCH
# This file is generated by ./scripts/bin/autogen
# Use that.
# typed: true

Opus::Require.on_autoload('Yabba::Dabba::Quuz')

class Yabba::Dabba::Quuz < AWS::String
end

Opus::Require.for_autoload(Yabba::Dabba::Quuz, "test/cli/autogen-autoloader/bar.rb")
